app/src/main/java/rkr/simplekeyboard/inputmethod/latin/LatinIME.java

```diff
@@ -527,6 +527,8 @@ void onStartInputViewInternal(final EditorInfo editorInfo, final boolean restart
             } else {
                 needToCallLoadKeyboardLater = false;
             }
+
+            mInputLogic.onUpdateSelection(editorInfo.initialSelStart, editorInfo.initialSelEnd);
```

Comment:
Selection is already saved in `InputLogic.mConnection.resetCachesUponCursorMoveAndReturnSuccess`. This will call the same flow for a second time. Also the retry case when `reloadTextCache` fails is not handled.